base/abstractdict.jl

```diff
@@ -516,7 +516,7 @@ convert(::Type{T}, x::T) where {T<:AbstractDict} = x
 
 function convert(::Type{T}, x::AbstractDict) where T<:AbstractDict
     h = T(x)
-    if length(h) != length(x)
+    if length(h)::Int != length(x)::Int
```

Comment:
I'm sure this is true in practice but it doesn't seem strictly correct to assert it. We don't know what `x` is.